### PR TITLE
Eri9600/distance measurement analysis

### PR DESCRIPTION
**Pull request by eri9600**
Distance Measurement Analysis sample as shown here: https://devtopia.esri.com/runtime/common-samples/issues/557

### CODE REVIEW

**Comment by eri9600** [comment]
@TADraeseke I am using  ArcGIS Version = 100.3.0-2062

File: java/distance-measurement-analysis/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 <resources>
-    <string name="app_name">distance-measurement-analysis</string>
+    <string name="app_name">Distance Measurement Analysis</string>
+    <string name="world_terrain_3d">http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer</string>
```
Comment (eri9600) [inline java/distance-measurement-analysis/src/main/res/values/strings.xml]: Do you recommend any other string names for these files? 
I looked at the links but I could not come up with another more descriptive string name.

File: java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    mSceneView.getAnalysisOverlays().add(analysisOverlay);
+
+    //initialize a distance measurement and add it to the analysis overlay
+    Point start = new Point(-4.494677, 48.384472, 24.772694, SpatialReferences.getWgs84());
+    Point end = new Point(-4.495646, 48.384377, 58.501115, SpatialReferences.getWgs84());
+    distanceMeasurement = new LocationDistanceMeasurement(start, end);
+    analysisOverlay.getAnalyses().add(distanceMeasurement);
+
+    // zoom to initial measurement
+    Camera camera = new Camera(start, 200.0, 0.0, 45.0, 0.0);
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java]: On my device, the end point is not in view when the device starts. Maybe increase the distance so both are in view on start?

File: java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    // add building layer
+    final String buildings = getResources().getString(R.string.buildings_tile);
+    ArcGISSceneLayer sceneLayer = new ArcGISSceneLayer(buildings);
+    scene.getOperationalLayers().add(sceneLayer);
+
+    // create analysis overlay and add it to scene
+    AnalysisOverlay analysisOverlay = new AnalysisOverlay();
+    mSceneView.getAnalysisOverlays().add(analysisOverlay);
+
+    //initialize a distance measurement and add it to the analysis overlay
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java]: space before initialize

File: java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    Point start = new Point(-4.494677, 48.384472, 24.772694, SpatialReferences.getWgs84());
+    Point end = new Point(-4.495646, 48.384377, 58.501115, SpatialReferences.getWgs84());
+    distanceMeasurement = new LocationDistanceMeasurement(start, end);
+    analysisOverlay.getAnalyses().add(distanceMeasurement);
+
+    // zoom to initial measurement
+    Camera camera = new Camera(start, 200.0, 0.0, 45.0, 0.0);
+    mSceneView.setViewpointCamera(camera);
+
+    // Initialize a list to contain the available units
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java]: lower case I

**Comment by TADraeseke** [comment]
Also, I rebased this PR to be against develop rather than master (since its using an internal build of 100.3 features)

**Comment by eri9600** [comment]
@TADraeseke I fixed the comments you made. What is the convention of view id's names regarding underscore or camelcase?

**Comment by TADraeseke** [comment]
@eri9600 - Don't know if there is a hard and fast rule. I tend to make file names (layout files, etc) and string ids with underscores. And then I tend to give views ids with camelcase. Both of these I do just because it tends to be the way Android studio automatically names stuff (ie activity_main for an xml file and textView for the id of a TextView)

File: java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    mSceneView.getAnalysisOverlays().add(analysisOverlay);
+
+    // initialize a distance measurement and add it to the analysis overlay
+    Point start = new Point(-4.494677, 48.384472, 24.772694, SpatialReferences.getWgs84());
+    Point end = new Point(-4.495646, 48.384377, 58.501115, SpatialReferences.getWgs84());
+    distanceMeasurement = new LocationDistanceMeasurement(start, end);
+    analysisOverlay.getAnalyses().add(distanceMeasurement);
+
+    // zoom to initial measurement
+    Camera camera = new Camera(start, 375.0, 0.0, 45.0, 0.0);
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/src/main/java/com/esri/arcgisruntime/sample/distancemeasurementanalysis/MainActivity.java]: Can we change the camera to allow both start and end points to be in view when the app starts? Something like:

Camera camera = new Camera(start, 300.0, 315.0, 60.0, 0.0);

File: java/distance-measurement-analysis/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+# Relevant API
+* AnalysisOverlay
+* LocationDistanceMeasurement
+* UnitSystem
+
+# Additional Information
+The `LocationDistanceMeasurement` analysis only performs planar distance calculations. This may not be appropriate for 
+large distances where the Earth's curvature needs to be taken into account.
+
+#Tags
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/README.md]: #### Tags

File: java/distance-measurement-analysis/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+1. Create an `AnalysisOverlay` and add it to your scene view's analysis overlay collection: `sceneView.getAnalysisOverlays().add(analysisOverlay)`. 
+1. Create a `LocationDistanceMeasurement`, specifying the `startLocation` and `endLocation`. These can be the same point to start with. Add the analysis to the analysis overlay: `analysisOverlay.getAnalyses().add(LocationDistanceMeasurement)`. The measuring line will be drawn for you between the two points. 
+1. The `measurementChanged` callback will fire if the distances change. You can get the new values for the `directDistance`, `horizontalDistance`, and `verticalDistance` from the `MeasurementChangedEvent` returned by the callback. The distance objects contain both a scalar value and a unit of measurement.
+
+# Relevant API
+* AnalysisOverlay
+* LocationDistanceMeasurement
+* UnitSystem
+
+# Additional Information
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/README.md]: ##

File: java/distance-measurement-analysis/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+start a new measurement. 
+
+# How It Works
+To measure distances with the `LocationDistanceMeasurement` analysis:
+
+1. Create an `AnalysisOverlay` and add it to your scene view's analysis overlay collection: `sceneView.getAnalysisOverlays().add(analysisOverlay)`. 
+1. Create a `LocationDistanceMeasurement`, specifying the `startLocation` and `endLocation`. These can be the same point to start with. Add the analysis to the analysis overlay: `analysisOverlay.getAnalyses().add(LocationDistanceMeasurement)`. The measuring line will be drawn for you between the two points. 
+1. The `measurementChanged` callback will fire if the distances change. You can get the new values for the `directDistance`, `horizontalDistance`, and `verticalDistance` from the `MeasurementChangedEvent` returned by the callback. The distance objects contain both a scalar value and a unit of measurement.
+
+# Relevant API
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/README.md]: ##

File: java/distance-measurement-analysis/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+so it doesn't interfere with other analyses like viewsheds.
+
+![Distance Measurement Analysis App](distance-measurement-analysis.png)
+
+# How To Use The Sample
+Choose an unit system for the measurement in the UI dropdown. Single tap any location on the screen to set the start point
+of measurement. Double tap and drag to set the end point of measurement. Single Tapping any new location will 
+start a new measurement. 
+
+# How It Works
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/README.md]: ##

File: java/distance-measurement-analysis/README.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+Measure 3D distances between two points in a scene.
+
+The distance measurement analysis allows you to add the same measuring experience found in ArcGIS Pro, City Engine,
+and the ArcGIS API for JavaScript to your app. You can set the unit system of measurement (metric or imperial) and 
+have the units automatically switch to one appropriate for the current scale. The rendering is handled internally 
+so it doesn't interfere with other analyses like viewsheds.
+
+![Distance Measurement Analysis App](distance-measurement-analysis.png)
+
+# How To Use The Sample
```
Comment (TADraeseke) [inline java/distance-measurement-analysis/README.md]: ##

**Comment by TADraeseke** [comment]
@eri9600 have proposed a slightly different initial camera to show both start and end point.

Also just need a bit of formatting on the README

**Comment by eri9600** [comment]
@TADraeseke I have made the suggested changes. Let me know if you find anything else.

**Comment by TADraeseke** [comment]
Looks good @eri9600! I just updated the screenshot to the new UI. Good to merge!

**Comment by eri9600** [comment]
@TADraeseke Thanks for the update. Merging now.

**Pull request merged**